File: src/extensibility/python-extensibility/nexus_extensibility/_data_model_utilities.py
from typing import Dict, Optional


def _get_representation_parameter_string(parameters: Optional[Dict[str, str]]) -> Optional[str]:
    
    if (parameters is None):
        return None

    serialized_parameters = (f"{parameter[0]}={parameter[1]}" for parameter in parameters.items())
    parameter_string = f"({','.join(serialized_parameters)})"

    return parameter_string

File: src/extensibility/rust-extensibility/src/data_model/data_model_utilities.rs
use super::representation::RepresentationParameters;

pub struct DataModelUtilities;

impl DataModelUtilities {
    pub fn get_representation_parameter_string(
        parameters: &Option<RepresentationParameters>,
    ) -> Option<String> {
        match parameters {
            Some(value) => {
                let serialized_parameters = value
                    .into_iter()
                    .map(|(key, value)| format!("{}={}", key, value))
                    .collect::<Vec<String>>();

                let parameters_string = format!("({})", serialized_parameters.join(","));

                Some(parameters_string)
            }
            None => None,
        }
    }
}
